app/src/main/java/com/babylon/wallet/android/presentation/ui/composables/assets/StakingTab.kt

```diff
@@ -122,42 +130,95 @@ private fun StakingSummary(
             )
         }
 
-        LaunchedEffect(summary) {
-            if (summary == null) {
+        LaunchedEffect(stakeSummary) {
+            if (stakeSummary == null) {
                 action.onStakesRequest()
             }
         }
 
+        val oneXrdPrice = remember(assetsViewData) {
+            val oneXrdPrice = (
+                assetsViewData.prices?.values?.find {
+                    (it as? AssetPrice.LSUPrice)?.oneXrdPrice != null
+                } as? AssetPrice.LSUPrice
+                )?.oneXrdPrice
+            oneXrdPrice
+                ?: (
+                    assetsViewData.prices?.values?.find {
+                        (it as? AssetPrice.StakeClaimPrice)?.oneXrdPrice != null
+                    } as? AssetPrice.StakeClaimPrice
+                    )?.oneXrdPrice
+        }
```

Comment:
maybe this could be in a function of AssetViewData?

Reply:
that's not really possible. Just tried and it failed because of the `oneXrdPrice` in line 145